src/backend/distributed/commands/multi_copy.c

```diff
@@ -3085,6 +3157,14 @@ CopyGetAttnums(TupleDesc tupDesc, Relation rel, List *attnamelist)
 					continue;
 				if (namestrcmp(&(att->attname), name) == 0)
 				{
+#if PG_VERSION_NUM >= PG_VERSION_12
+					if (att->attgenerated)
+						ereport(ERROR,
+								(errcode(ERRCODE_INVALID_COLUMN_REFERENCE),
+								 errmsg("column \"%s\" is a generated column",
+										name),
+								 errdetail("Generated columns cannot be used in COPY.")));
```

Comment:
This seems to not be covered in the tests.

Does vanilla postgres support this? What would it entail for use to support it?

Reply:
This is copied from vanilla postgres